crates/bevy_ptr/README.md

```diff
@@ -6,8 +6,104 @@
 [![Docs](https://docs.rs/bevy_ptr/badge.svg)](https://docs.rs/bevy_ptr/latest/bevy_ptr/)
 [![Discord](https://img.shields.io/discord/691052431525675048.svg?label=&logo=discord&logoColor=ffffff&color=7389D8&labelColor=6A7EC2)](https://discord.gg/bevy)
 
-The `bevy_ptr` crate provides low-level abstractions for working with pointers in a more safe way than using rust's raw pointers.
+Pointers in computer programming, according to Wikipedia, are "objects in many programming languages that stores a memory address". 
+They're a fundamental building block for constructing more complex data structures.
 
-Rust has lifetimed and typed references (`&'a T`), unlifetimed and typed references (`*const T`), but no lifetimed but untyped references.
-`bevy_ptr` adds them, called `Ptr<'a>`, `PtrMut<'a>` and `OwningPtr<'a>`.
-These types are lifetime-checked so can never lead to problems like use-after-frees and must always point to valid data.
+They're also *the* definitive source of memory safety bugs. You can dereference a null pointer. Access a pointer after the underlying 
+memory has been freed. Ignore type safety and misread or mutate the underlying memory improperly.
+
+Rust is a programming language that heavily relies on its types to enforce correctness, and by proxy, memory safety. As a result, 
+Rust has an entire zoo of types for working with pointers, and a graph of safe and unsafe conversions to among them to make working 
+with them safer.
+
+`bevy_ptr` is a crate that attempts to bridge the gap between the full blown unsafety of `*mut ()` and the safe `&'a T`, allowing users 
+to build progressively to choose what invariants to uphold.
+
+## How to build a Borrow (from scratch)
```

Comment:
```suggestion
## How to Build a Borrow (From Scratch)
```

I think titles should be consistent in their capitalization. Later headings capitalize important words, so I think this should do the same.